sdk/docs/manually-written/sdk/component-howtos/smart-contracts/flags.rst

```diff
@@ -1,112 +1,146 @@
 .. Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
+
+.. _howto-daml-compiler:
+
+The Daml Compiler
+#################
+
+At the core of the Daml toolchain lies the Daml Compiler, also known as `damlc`.
+Its primary usage is to compile Daml source code (with ``.daml`` extension) into
+a lower-level language called Daml-LF. :ref:`Daml-Lf<daml-lf>` is the language
+that canton participants can evaluate.
+
+When building, recommended via :subsiteref:`DPM<dpm>` (i.e. trough ``dpm
+build``), or alternatively, by calling ``damlc`` directly, the Daml Compiler
+goes through the following stages:
+
+
+#. The input ``.daml`` :ref:`files<glos-daml-source-files>` are type-checked (i.e. it is verified if they are
+   well-formed or not)
+#. The type-checked ```.daml`` code is transformed to
+   :ref:`Daml-LF<daml-lf-intro>`.
+#. The Daml-LF is packaged together with its dependency in a ``.dar``
+   :ref:`file<dar-file-dalf-file>`, which essentially is an archive of Daml-LF
+   files with metadata.
+
+Build files can be cleaned using ``dpm clean``.
+
+On top of that, the Daml Compiler offers testing functionality, either by
+calling ``dpm build`` to obtain a ``.dar`` to be used in combination with
```

Comment:
```suggestion
The Daml Compiler also offers testing functionality. It can be invoked either by
first calling ``dpm build`` to obtain a ``.dar`` which is then used in a call to
```